cmd/ipfs-cluster-service/main.go

```diff
@@ -632,3 +641,40 @@ func getStateManager() cmdutils.StateManager {
 	checkErr("creating state manager", err)
 	return mgr
 }
+
+func assignRandomPorts(multiAddrs []*ma.Multiaddr) {
+	for _, m := range multiAddrs {
+		var withRandom ma.Multiaddr
+		ma.ForEach(*m,
+			func(c ma.Component) bool {
+				code := c.Protocol().Code
+				var port int
+				if code == ma.P_TCP || code == ma.P_UDP {
+					ln, err := net.Listen("tcp", "[::]:0")
```

Comment:
why ipv6?

Reply:
Should it really matter as long as it creates a random available port? I have changed this though.